### PR TITLE
Add JSON schema form visual improvements

### PR DESCRIPTION
**Pull request by gusevda**
### What does this PR do?

Visual improvements for hierarchical object and array fields according to the [design specs](https://www.figma.com/file/DsMPSXboZwWQjEryN5Y8wv/Happa-generated-hierarchical-form?node-id=0%3A1&t=RSAys5SdSeNqhUUC-0).

### How does it look like?

Screenshot 2023-01-26 at 14 15 53

### Any background context you can provide?
Towards https://github.com/giantswarm/roadmap/issues/1181.

### CODE REVIEW

**Comment by marians** [comment]
It would be nice to have an example in our test schema where array items are objects with a title. Like this:

```json
                "arrayOfObjectsWithTitle": {
                    "items": {
                        "properties": {
                            "age": {
                                "type": "number"
                            },
                            "name": {
                                "type": "string"
                            }
                        },
                        "type": "object",
                        "title": "Array item object",
                        "description": "Description of array item object"
                    },
                    "title": "Array of objects with title",
                    "type": "array"
                },
```

**Comment by marians** [comment]
Why is there an asterisk on the generated object title `arrayOfObjects-0` here?

image

**Comment by gusevda** [comment]
> Why is there an asterisk on the generated object title `arrayOfObjects-0` here?

The library mark array items as required for some reason. I would need to check what we can do about it.

**Comment by gusevda** [comment]
> It would be nice to have an example in our test schema where array items are objects with a title.

I added an example into the test schema. When an array item has custom title, it's being used without index postfix. If we still want to have index added, I can look into it. It's better to do it in a separate ticket, I think.

File: src/components/UI/JSONSchemaForm/ArrayFieldItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  hasMoveUp,
+  hasRemove,
+  hasToolbar,
+  readonly,
+  onDropIndexClick,
+  onReorderClick,
+  children,
+}) => {
+  return (
+    <ArrayFieldItemW direction='row'>
```
Comment (kuosandys) [inline src/components/UI/JSONSchemaForm/ArrayFieldItem/index.tsx:43]: Just a small suggestion - could we have a more descriptive name than `ArrayFieldItemW` ? I assumed the `W` stands for wrapper, but I wasn't sure if the abbreviation was intentional

**Review by kuosandys — APPROVED** [review]
Works well for me 👍 My comment above about the naming shouldn't be blocking

File: src/components/UI/JSONSchemaForm/ArrayFieldItem/index.tsx (hunk 1 above)
Comment (gusevda) [reply src/components/UI/JSONSchemaForm/ArrayFieldItem/index.tsx:43]: Thank you for noticing, it's definitely unintentional. Will fix it.

**Review by marians — APPROVED** [review]
Good improvement as it is!

**Comment by marians** [comment]
Referencing the original issue: https://github.com/giantswarm/roadmap/issues/1181

**Pull request merged**